platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -132,14 +139,14 @@ void printDiagnosticCounts() {
      * @param kind the kind of diagnostic (error, or warn)
      * @param number the total number of diagnostics of the given kind
      */
-    private static void printDiagnosticCount(Log logger, String kind, int number) {
+    private static String diagnosticCount(Log logger, String kind, int number) {
```

Comment:
💭 Could this and `diagnosticCounts` return `Optional<String>` to allow for a nicer `ifPresent` usage when checking to print?

Reply:
It looks nicer with the proposed changes.
However I wonder if it makes sense to create all those objects just for two nullable variables. Here the difference is minuscule, but a widespread use of this pattern might add up to something significant. IIRC the recommendation for  `Optional` usage is only for API calls, `null` values are fine for local variables and private method return values.